Framework/DataHandling/src/SaveGDA.cpp

```diff
@@ -0,0 +1,260 @@
+const std::vector<std::string> SaveGDA::seeAlso() const {
+  return {"SaveBankScatteringAngles", "AlignDetectors"};
+}
+
+const std::string SaveGDA::category() const {
+  return "DataHandling\\Text;Diffraction\\DataHandling";
+}
+
+const std::string SaveGDA::PROP_OUTPUT_FILENAME = "Filename";
+
+const std::string SaveGDA::PROP_INPUT_WS = "InputWorkspace";
+
+const std::string SaveGDA::PROP_PARAMS_FILENAME = "GSASParamFile";
+
+const std::string SaveGDA::PROP_GROUPING_SCHEME = "GroupingScheme";
+
+void SaveGDA::init() {
+  declareProperty(Kernel::make_unique<WorkspaceProperty<WorkspaceGroup>>(
+                      PROP_INPUT_WS, "", Kernel::Direction::Input),
+                  "A GroupWorkspace where every sub-workspace is a "
+                  "single-spectra focused run corresponding to a particular "
+                  "bank");
+
+  const static std::vector<std::string> outExts{".gda"};
+  declareProperty(Kernel::make_unique<FileProperty>(
+                      PROP_OUTPUT_FILENAME, "", FileProperty::Save, outExts),
+                  "The name of the file to save to");
+
+  const static std::vector<std::string> paramsExts{".ipf", ".prm", ".parm",
+                                                   ".iprm"};
+  declareProperty(
+      Kernel::make_unique<FileProperty>(PROP_PARAMS_FILENAME, "",
+                                        FileProperty::Load, paramsExts),
+      "GSAS calibration file containing conversion factors from D to TOF");
+
+  declareProperty(
+      Kernel::make_unique<Kernel::ArrayProperty<int>>(PROP_GROUPING_SCHEME),
+      "An array of bank IDs, where the value at element i is the "
+      "ID of the bank in " +
+          PROP_PARAMS_FILENAME + " to associate spectrum i with");
+}
+
+void SaveGDA::exec() {
+  const std::string filename = getProperty(PROP_OUTPUT_FILENAME);
+  std::ofstream outFile(filename.c_str());
+
+  if (!outFile) {
+    throw Kernel::Exception::FileError("Unable to create file: ", filename);
+  }
+
+  outFile << std::fixed << std::setprecision(0) << std::setfill(' ');
+
+  const API::WorkspaceGroup_sptr inputWS = getProperty(PROP_INPUT_WS);
+  const auto numBanks = inputWS->getNumberOfEntries();
+  const auto calibParams = parseParamsFile();
+  const std::vector<int> groupingScheme = getProperty(PROP_GROUPING_SCHEME);
+
+  for (int i = 0; i < numBanks; ++i) {
+    const auto ws = inputWS->getItem(i);
+    const auto matrixWS = boost::dynamic_pointer_cast<MatrixWorkspace>(ws);
+
+    const auto &d = matrixWS->x(0);
+    const auto &bankCalibParams = calibParams[groupingScheme[i] - 1];
+
+    std::vector<double> tofScaled;
+    tofScaled.reserve(d.size());
+    std::transform(d.begin(), d.end(), std::back_inserter(tofScaled),
+                   [&bankCalibParams](const double dVal) {
+                     return (dVal * bankCalibParams.difa +
+                             dVal * dVal * bankCalibParams.difc +
+                             bankCalibParams.tzero) *
+                            32;
```

Comment:
Document this magic value in a variable, even if it itself is magic (does that make sense?)